api/src/main/java/org/apache/iceberg/metrics/ScanReport.java

```diff
@@ -102,40 +113,304 @@ public Builder withSnapshotId(long newSnapshotId) {
       return this;
     }
 
-    public Builder withFilter(Expression newFilter) {
-      this.filter = newFilter;
-      return this;
-    }
-
     public Builder withProjection(Schema newProjection) {
       this.projection = newProjection;
       return this;
     }
 
     public Builder fromScanMetrics(ScanMetrics newScanMetrics) {
-      this.scanMetrics = newScanMetrics;
+      this.scanMetrics = ScanMetricsResult.fromScanMetrics(newScanMetrics);
+      return this;
+    }
+
+    public Builder fromScanMetricsResult(ScanMetricsResult newScanMetricsResult) {
+      this.scanMetrics = newScanMetricsResult;
       return this;
     }
 
     public ScanReport build() {
       Preconditions.checkArgument(null != tableName, "Invalid table name: null");
-      Preconditions.checkArgument(null != filter, "Invalid expression filter: null");
       Preconditions.checkArgument(null != projection, "Invalid schema projection: null");
       Preconditions.checkArgument(null != scanMetrics, "Invalid scan metrics: null");
-      return new ScanReport(tableName, snapshotId, filter, projection, scanMetrics);
+      return new ScanReport(tableName, snapshotId, projection, scanMetrics);
+    }
+  }
+
+  /** A serializable version of a {@link Timer} that carries its result. */
+  public static class TimerResult implements Serializable {
```

Comment:
Why implement `Serializable`? I thought the requirement was to make it serializable with a JSON parser.

Reply:
removed